zyte_test_websites/ecommerce/items.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from aiohttp import web
+
+
+@dataclass(frozen=True)
+class ProductCategory:
+    id: int
+    name: str
+    parent_category_id: int | None
+    subcategories: dict[int, ProductCategory]
+    products: dict[int, Product]
+
+    def to_dict(self) -> dict[str, Any]:
+        return {
+            "id": self.id,
+            "name": self.name,
+            "parent": self.parent_category_id,
+        }
+
+
+@dataclass(frozen=True)
+class Product:
```

Comment:
I wonder if there is a better module name; I was expecting to see definitions of Scrapy-like items returned by ecommerce page objects in such module, but it seems there are some internal data structures used by the website, which users shouldn't worry about (?).

Reply:
Yeah, `app` and `extraction` contain things useful to the user while `items` and `views` are only used by the website itself. I don't know if it's worth it to separate these somehow. This is also why it's `extraction` and not `pages`, as `pages` sounds like something used by the website, not something related to web-poet.